building_dialouge_webapp/heat/tables.py

```diff
@@ -22,36 +62,108 @@ class Table:
+            table_data[self.translate_column(scenario)] = [
+                scenario_data.get("investment", 0),
+                scenario_data.get("contribution", 0),
             ]
 
-        return pd.DataFrame(table_data)
+        dataframe = pd.DataFrame(table_data)
+        sum_row = [self.translate_column("Summe"), *dataframe.iloc[:, 1:].sum().tolist()]
+        dataframe.loc[len(dataframe)] = sum_row
+        for col in dataframe.columns[1:]:
+            dataframe[col] = dataframe[col].apply(format_currency)
+        return dataframe
 
-    def to_html(self):
+    def to_html(self, title="investment_table"):
         dataframe = self.generate_table_data()
+        return super().to_html(dataframe, title)
 
-        html_table = dataframe.to_html(classes="table", escape=False, index=False)
 
-        html_table = html_table.replace(
-            '<table border="1" class="dataframe table">',
-            '<table class="consumption_table">',
+class TabularTable(Table):
+    translations = {
+        **Table.translations,
+        "air_heat_pump": "Luft-Wärmepumpe",
+        "thermal_storage": "Wärmespeicher",
+        "heat_meter": "Wärmemengenzähler",
+        "heating_system_pump": "Pumpe des Heizsystems",
+        "ventilation_system": "Lüftungsanlage",
+        "insulate_outer_facade": "Außenfassade dämmen",
+        "insulate_roof": "Dach dämmen",
+    }
+
+    def generate_table_data(self, scenario_data):
+        investments = scenario_data.get("investments", {})
+        subsidies = scenario_data.get("subsidies", {})
+
+        dataframe_investments = pd.DataFrame(
+            {
+                "Investitionskosten": [self.translations.get(k, k) for k in investments],
+                "": [format_currency(v) for v in investments.values()],
+            },
+        )
+
+        dataframe_subsidies = pd.DataFrame(
+            {
+                "Zuschüsse": list(subsidies),
+                "": [format_currency(v) for v in subsidies.values()],
+            },
         )
-        html_table = html_table.replace("<th>procedure</th>", '<th class="procedure">Maßnahme</th>')
-        html_table = html_table.replace("<th>Szenario 1</th>", '<th class="scenario1">Szenario 1</th>')
-        html_table = html_table.replace("<th>Szenario 2</th>", '<th class="scenario2">Szenario 2</th>')
-        html_table = html_table.replace("<th>Szenario 3</th>", '<th class="scenario3">Szenario 3</th>')
 
-        return html_table.replace("<td>nicht ausgewählt</td>", '<td class="no-entry">nicht ausgewählt</td>')
+        total_cost = sum(investments.values()) - sum(subsidies.values())
+        dataframe_sum = pd.DataFrame(
+            {
+                "Summe": [""],
+                "": [format_currency(total_cost)],
+            },
+        )
+
+        return dataframe_investments, dataframe_subsidies, dataframe_sum
+
+    def generate_html_table(self, scenario_data, scenario_class, title="summary_table"):
```

Comment:
This should be refactored into `to_html` as well in order to make it consistent with `table` classes from above.
I see that this function takes additional attribute `scenario_class`, but could this be part of `scenario_data` as well - or maybe we could adapt `to_html` function of parent class to support this feature as well, if this makes sense?